### ISSUE TITLE
Entry.log modifies writes Level, not reusable. 

### ISSUE THREAD

**Issue by kojustin** [sirupsen/logrus#393]
Entry.log modifies "entry.Level" to whatever level the current message is being logged at. This means that once an Entry has logged a more severe message it can no longer log a less severe message. The documentation says

```
// An entry is the final or intermediate Logrus logging entry. It contains all
// the fields passed with WithField{,s}. It's finally logged when Debug, Info,
// Warn, Error, Fatal or Panic is called on it. These objects can be reused and
// passed around as much as you wish to avoid field duplication.
```

If the object modifies itself it can't be reused. Here's a test program. I expect to see 5 log messages.

```
package main

import "github.com/Sirupsen/logrus"

func main() {
    logrus.SetLevel(logrus.DebugLevel)
    entry := logrus.WithField("thread", "main")

    // Less serious to more serious.
    entry.Debugf("debug entry.Level=%d", entry.Level)
    entry.Infof("info entry.Level=%d", entry.Level)
    entry.Errorf("errorf entry.Level=%d", entry.Level)

    // more serious to less serious.
    entry.Infof("Info entry.Level=%d", entry.Level)
    entry.Debugf("debug entry.Level=%d", entry.Level)

}
```

This is the output I see:

```
$ go run main.go 
DEBU[0000] debug entry.Level=5                           thread=main
INFO[0000] info entry.Level=5                            thread=main
ERRO[0000] errorf entry.Level=4                          thread=main
```

**Comment by zstyblik** [comment]
This seems to be fixed in master branch.

**Comment by rdtr** [comment]
if so, maybe we can close this @kojustin ?

**Comment by kojustin** [comment]
ok sue

**Comment by kojustin** [comment]
Actually, has this fix been released yet?

**Comment by rdtr** [comment]
I haven't checked the issue in detail, but isn't this the commit that @zstyblik mentioned?
https://github.com/sirupsen/logrus/commit/53cbb9dc6df4a5ae2a1c85ff3a265b832d7667db

**Comment by zstyblik** [comment]
Sorry guys, I've just tested the code provided by @kojustin and didn't look for particular git commit. :|

**Comment by kojustin** [comment]
ok verified fixed at v0.11.0
